### DISCUSSION TITLE
How to build & install libvips only with meson?

### DISCUSSION THREAD

**Discussion by der-eismann** [libvips/libvips#3227] [Q&A]
We have a project that depends on the vips shared library only, so we don't want to build the tools or c++ stuff to not waste energy. At the moment we checkout the project and then do
```
./configure  ...
cd libvips && make -j8 && make DESTDIR=/ install
```

However with meson this seems much more complicated. I now have this setup:
```
meson setup build
meson compile libvips/vips -C build && \
meson compile libvips/vips-heif -C build && \
meson compile libvips/vips-jxl -C build && \
meson install -C build --no-rebuild
```
However if I do this the install step complains that some parts that were not built can't be found, e.g.
```
ERROR: File 'cplusplus/libvips-cpp.so.42.15.3' could not be found
```
When I use `meson install --tags=runtime` it also still wants to install the C++ stuff. Any hints how to achieve the previous behavior? Could we maybe expand meson with some custom tags?

**Comment by jcupitt** [comment]
Hi @der-eismann,

Sure, a PR with some extra options would be useful. Have a look at the `-Dexamples` option if you need a guide.

There are some now -- have a look in `meson_options.txt`. You could build with (for example):

```
meson setup build --prefix ~/vips \
    -Ddeprecated=false \
    -Dexamples=false \
    -Dintrospection=false \
    -Danalyze=false \
    -Dradiance=false \
    -Dppm=false 
```

Then:

```
meson compile -C build
meson install -C build
```

**Reply by jcupitt** [reply]
... I added a `-Dcplusplus=false` option to disable build of the C++ API: https://github.com/libvips/libvips/commit/821e2165b91fb13269123e306c38d2e2e7656d83

Disabling the tools would be harder: you'd need to stop most of the tests from running too.

**Reply by eli-schwartz** [reply]
You could add the tools as `depends: [tools, that, are, needed]` to the tests that need them. When the tools are disabled, define them as a disabler() instead -- then the tests that depend on those tools end up disabled as well.
